Cargo.toml

```diff
@@ -55,6 +55,8 @@ members = [
     "transports/websocket",
     "transports/webtransport-websys",
     "wasm-tests/webtransport-tests",
+    "wasm-tests/webrtc",
+    "wasm-tests/webrtc/server",
```

Comment:
I'd suggest to prioritize adding interop tests over these wasm-tests and only use these to test certain APIs in a more fine granular way. The interop tests act as a kind of smoke test that tell you whether or not things are basically working which I think is more valuable to start with :)

Once we have that, we can think of more test cases that we might want to cover although I am not sure there are that many, especially if we create the `libp2p-webrtc-framed-stream` crate as suggested above which we can unit-test with something like `futures-ringbuf`.

Reply:
Yes I would be happy to take the interop route, I am at the point where I need to start running more tests anyway. Right now we can `Dial`, but I am nearly ready to test more.

I've taken a look, but I am unfamiliar with where these interop tests are, how to add to them/use them, or how Transports are currently tested. Could you add some links to point me in the right direction please? I understand you know them well and could save me a ton of time trying to figure it all out from scratch

Reply:
Of course!

This is the transport you want to extend: https://github.com/libp2p/rust-libp2p/blob/ceccb8e9c0e4b34ea651109c1b325f62ebdb69aa/interop-tests/src/arch.rs#L180

You need to add `webrtc-direct` / `webrtc` here too for it to be picked up: https://github.com/libp2p/rust-libp2p/blob/ceccb8e9c0e4b34ea651109c1b325f62ebdb69aa/interop-tests/chromium-ping-version.json#L4

(Are you implementing both btw? Or just one of the two?)

Here are some docs on how to run a subset of these tests locally: https://github.com/libp2p/test-plans/tree/master/multidim-interop#running-locally

If you don't want to bother with that, you can also just push commits to CI and wait although that might be rather painful in regards to iteration time 😬

Reply:
Done (well, started). I'm still finding this useful for tracing the route, but it doesn't necessarily need to make it into the final Pull Request